### PR TITLE
Add saml_group_mappings resource

### PR DESCRIPTION
**Pull request by paivagustavo**
Add a new `saml_group_mappings` resource to manage an organization's SAML Group Mappings.

### CODE REVIEW

File: docs/resources/saml_group_mappings.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+- `attribute_key` (String) Attribute Key to match against the user's SAML attributes.
+- `attribute_value` (String) Attribute Value to match against the user's SAML attributes
+
+
+<a id="nestedblock--mapping--roles"></a>
+### Nested Schema for `mapping.roles`
+
+Required:
+
+- `organization_role` (String) Organization Role. Only 'Organization Editor' and 'Organization Viewer' are supported.
```
Comment (devinyoungcs) [inline docs/resources/saml_group_mappings.md:81]: Should Organization Restricted Member be a valid option in this list as well?

File: lightstep/resource_saml_group_mappings.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+							Type:        schema.TypeList,
+							Required:    true,
+							MaxItems:    1,
+							Description: "Roles to assign to the user if the match is successful. ",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"organization_role": {
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "Organization Role. Only 'Organization Editor' and 'Organization Viewer' are supported.",
```
Comment (devinyoungcs) [inline lightstep/resource_saml_group_mappings.go:60]: (duplicate comment from above): add Org Restricted Member to this list too

File: lightstep/resource_saml_group_mappings.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+	for _, rawMapping := range rawMappings.([]any) {
+		var mapping client.SAMLGroupMapping
+
+		rawMappingMap := rawMapping.(map[string]any)
+
+		rawMatch, ok := rawMappingMap["match"]
+		if !ok {
+			return mappings, errors.New("missing required field 'match'")
+		}
+		rawNatchSlice := rawMatch.([]any)
```
Comment (devinyoungcs) [inline lightstep/resource_saml_group_mappings.go:178]: rawNatchSlice -> rawMatchSlice

File: docs/resources/saml_group_mappings.md (hunk 1 above)
Comment (paivagustavo) [reply docs/resources/saml_group_mappings.md]: yeah, nice catch! Updated it on both places, thank you!

File: lightstep/resource_saml_group_mappings.go (hunk 3 above)
Comment (paivagustavo) [reply lightstep/resource_saml_group_mappings.go]: ops, nice :eyes:, fixed it! :)

**Review by devinyoungcs — APPROVED** [review]
🔥

**Comment by karticus** [comment]
Merging as part of saml release

**Pull request merged**